include/cooperative_perception/angle.hpp

```diff
@@ -0,0 +1,127 @@
+/*
+ * Copyright 2022 Leidos
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#ifndef COOPERATIVE_PERCEPTION_ANGLE_HPP
+#define COOPERATIVE_PERCEPTION_ANGLE_HPP
+
+#include <complex>
+#include <units.h>
+#include <boost/math/constants/constants.hpp>
+#include "cooperative_perception/utils.hpp"
+
+namespace cooperative_perception
+{
+/**
+ * @brief Angle type for representing and manipulating angles
```

Comment:
I would suggest adding a comment similar to this
https://github.com/usdot-fhwa-stol/cooperative-perception-core/pull/17#issuecomment-1371458930

Which explains the internal design decision of the complex plane